tests/console/check_package_version.pm

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Check package version only for sles12sp5 migration scenarios
+# Maintainer: Yutao Wang <yuwang@suse.com>
+
+use strict;
+use warnings;
+use version;
+use base 'installbasetest';
+use testapi;
+use utils 'systemctl', 'zypper_call';
+
+my @packages = qw(autofs openscap augeas freeradius-server gpgme libgpg-error0); #apache2-mod_auth_openidc);
+my @packagev = qw(5.1.5 1.3.0 1.0.0 3.0.18 1.9.0 1.17);# 0.0.0);
+
+my @packages_s390x = qw(libnuma-devel);
+my @packagev_s390x = qw(2.0.12);
+
+sub search_packages {
+    my ($package) = @_;
+    zypper_call("se -s $package");
+}
+
+sub cmp_version {
+    my ($old, $new) = @_;
+    my @newv = split(/-/, $new);
+    my $v1 = version->parse($old);
+    my $v2 = version->parse($newv[0]);
+    return $v1 <= $v2;
+}
+
+sub cmp_result {
+    my ($pcks, $pckv) = @_;
+    search_packages($pcks);
+    my $output = script_output("zypper se -s $pcks | grep -w $pcks | head -1 | awk -F '|' '{print \$4}'", 80);
+    if ($output ne '' && !cmp_version($pckv, $output)) {
+         record_soft_failure("The $pcks version is $output, but request is $pckv");
```

Comment:
I think we should fail here . first of all it is look like main check of the test. also golden rule is to use softfail **only** with known bug id , so please add bug here or replace it with failure

Reply:
This part, My opinion is just not impact other packages' version checking.  If one package version is not correct, it can keep to check others version.  Then we can check all packages' version.

Reply:
main purpose of softfail status is to cover cases when there is known bug which have not been fixed for whatever reason but we want to continue with testing nevertheless. So basically when reviewer see test case in status softfail expected judgment about it would be - "this test case has no issues except already detected ". 
Here I see abuse of this logic, which would lead to situation that test case would actually fail ( there will be packages with version mismatch) but this will be ignored because normally test reviewers not look inside softfailed test cases because they expect them not to have anything new. 
Regarding scenario which you want to achieve ( keep checking packages till the end even when first showing version mismatch ) - this can be achieved by if function `cmp_result` will not fail or softfail but will just return result (true/false) which you can collect and fail whole test case at the end after ALL packages have been checked

Reply:
I changed it to 'record_info' to record the failed information.